tests/textcase/lsp/lsp_spec.lua

```diff
@@ -1 +1,62 @@
-describe("LSP", function() end)
+local test_helpers = require("tests.test_helpers")
+local textcase = require("textcase")
+
+local cur_dir = vim.fn.expand("%:p:h")
+
+---@param path string
+function M.read_file(path)
+  local fd = vim.loop.fs_open(path, "r", 438)
+  local fstat = vim.loop.fs_fstat(fd)
+  local contents = vim.loop.fs_read(fd, fstat.size, 0)
+  vim.loop.fs_close(fd)
+  return contents
+end
+
+function M.waitFor(max_time, callback)
+  local max_seconds = max_time
+  local curr_seconds = 0
+  local task_finished = false
+  while curr_seconds < max_seconds and not task_finished do
+    curr_seconds = curr_seconds + 0.5
+    task_finished = callback()
+    vim.wait(500, function() end)
+  end
+end
+
+describe("LSP", function()
+  describe("Rename", function()
+    before_each(function()
+      textcase.setup({})
+    end)
+
+    it("Should be triggered on keybinding", function()
+      local path = cur_dir .. "/tests/textcase/lsp/fixtures/component-camel-case.tsx"
+      local cmd = " silent exe 'e " .. path .. "'"
+      vim.cmd(cmd)
+      vim.bo.filetype = "typescriptreact"
+      -- allow tsserver start
+
+      local ts_server_started = false
+      M.waitFor(30, function()
+        vim.cmd("LspInfo")
+        ts_server_started = not not vim.fn.search("1 client(s) attached to this buffer")
+        test_helpers.execute_keys("q")
+        return ts_server_started
+      end)
+
+      assert.is.truthy(ts_server_started)
+      vim.wait(1000, function() end)
```

Comment:
1 second seems a bit long. I think 100ms should be enough.

Reply:
I wish, even 500ms is not enough. The issue is in the `pipeline` and `act`. `just tests` runs fine with 50ms
I don't even understand what this await is for.

In the previous `wait_for`, it already makes sure the LSP was loaded and it has closed the LspInfo window. So, there is no apparent reason for this `vim.wait`, but if it is missing the pipeline tests does not work

Reply:
Ok :/ good to know! Can you add a comment for this then?